### PR TITLE
Update Genrich module input tuple and argument

### PR DESCRIPTION
**Pull request by samuelruizperez**
This PR:

- Joins `treatment` and `control` in tuple of input channels.
- Makes it possible to input a list of `treatment` or `control` files, convert it to a comma-separated list, and analyse biological replicates simultaneously (see [Genrich#multiple-replicates](https://github.com/jsh58/Genrich#multiple-replicates)). A test using two treatments as input was added.
- Adds `-y` `(Keep unpaired alignments)` as the default when `meta.single_end:true` (see [Genrich#unpaired-alignments](https://github.com/jsh58/Genrich#unpaired-alignments) and [Genrich/issues/#77](https://github.com/jsh58/Genrich/issues/77#issuecomment-876073734)).

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] Follow the naming conventions.
- [x] Follow the input/output options guidelines.
- [x] Follow the parameters requirements.
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/genrich/main.nf [hunk 1]
```diff
@@ -8,34 +8,36 @@ process GENRICH {
...
-    def args = task.ext.args ?: ''
-    def prefix = task.ext.prefix ?: "${meta.id}"
-    def control    = control_bam    ? "-c $control_bam"               : ''
-    def blacklist  = blacklist_bed  ? "-E $blacklist_bed"             : ""
-    def pvalues    = save_pvalues   ? "-f ${prefix}.pvalues.bedGraph" : ""
-    def pileup     = save_pileup    ? "-k ${prefix}.pileup.bedGraph"  : ""
-    def bed        = save_bed       ? "-b ${prefix}.intervals.bed"    : ""
+    def args       = task.ext.args   ?: ""
+    def prefix     = task.ext.prefix ?: "${meta.id}"
+    def layout     = (!args.contains("-y") && meta.single_end) ? "-y" : ""
```
Comment (SPPearce) [inline modules/nf-core/genrich/main.nf:32]: What is this doing?

File: modules/nf-core/genrich/main.nf [hunk 2]
```diff
@@ -8,34 +8,36 @@ process GENRICH {
...
+    def treatment  = treatment_bam   ? "-t ${treatment_bam.sort().join(',')}"   : ""
```
Comment (SPPearce) [inline modules/nf-core/genrich/main.nf:33]: Why do you sort here?

**Review by SPPearce — COMMENTED** [review]
Generally I think this is fine, but I've never used the tool.

Question: Do we need all these value channels here? They only determine whether or not to save output, which could be controlled by ext.args.

File: modules/nf-core/genrich/main.nf (hunk 1 above)
Comment (JoseEspinosa) [reply modules/nf-core/genrich/main.nf:32]: Yes, would like also to know

File: modules/nf-core/genrich/main.nf (hunk 2 above)
Comment (JoseEspinosa) [reply modules/nf-core/genrich/main.nf:33]: Yes, is it trying to maintain the correspondence between control and treatment files?

**Review by JoseEspinosa — COMMENTED** [review]
Just drop two comments essentially the same as @SPPearce, would be great if you could address them in order to move https://github.com/nf-core/atacseq/pull/331 on

File: modules/nf-core/genrich/main.nf (hunk 1 above)
Comment (samuelruizperez) [reply modules/nf-core/genrich/main.nf]: Single-end data can only be analyzed if unpaired alignments are kept (`-y` or `-w <int>`) so it defaults to `-y` if neither is already in `args` ([https://github.com/jsh58/Genrich/issues/77#issuecomment-876073734](https://github.com/jsh58/Genrich/issues/77#issuecomment-876073734)).

File: modules/nf-core/genrich/main.nf (hunk 2 above)
Comment (samuelruizperez) [reply modules/nf-core/genrich/main.nf]: > Yes, is it trying to maintain the correspondence between control and treatment files?

That was my intention, but I don't think it's necessary (?)

**Comment by samuelruizperez** [comment]
> Question: Do we need all these value channels here? They only determine whether or not to save output, which could be controlled by ext.args.

 Removed the optional value channels.

**Comment by SPPearce** [comment]
I need to review properly on my computer not my phone, but looks better at first pass.

File: modules/nf-core/genrich/main.nf [hunk 3]
```diff
@@ -8,54 +8,43 @@ process GENRICH {
...
-    tuple val(meta), path("*pvalues.bedGraph"), optional:true, emit: bedgraph_pvalues
-    tuple val(meta), path("*pileup.bedGraph") , optional:true, emit: bedgraph_pileup
-    tuple val(meta), path("*intervals.bed")   , optional:true, emit: bed_intervals
-    tuple val(meta), path("*duplicates.txt")  , optional:true, emit: duplicates
-    path "versions.yml"                                      , emit: versions
+    tuple val(meta), path("*.narrowPeak")                     , emit: peak
+    path "versions.yml"                                       , emit: versions
+
+    tuple val(meta), path("*.pvalues.bedGraph"), optional:true, emit: bedgraph_pvalues
+    tuple val(meta), path("*.pileup.bedGraph"),  optional:true, emit: bedgraph_pileup
```
Comment (SPPearce) [inline modules/nf-core/genrich/main.nf:19]:
```suggestion
    tuple val(meta), path("*.pileup.bedGraph") ,  optional:true, emit: bedgraph_pileup
```

File: modules/nf-core/genrich/main.nf [hunk 4]
```diff
@@ -8,54 +8,43 @@ process GENRICH {
...
+    tuple val(meta), path("*.intervals.bed"),    optional:true, emit: bed_intervals
```
Comment (SPPearce) [inline modules/nf-core/genrich/main.nf:20]:
```suggestion
    tuple val(meta), path("*.intervals.bed") ,    optional:true, emit: bed_intervals
```

File: modules/nf-core/genrich/main.nf [hunk 5]
```diff
@@ -8,54 +8,43 @@ process GENRICH {
...
+    tuple val(meta), path("*.duplicates.txt"),   optional:true, emit: duplicates
```
Comment (SPPearce) [inline modules/nf-core/genrich/main.nf:21]:
```suggestion
    tuple val(meta), path("*.duplicates.txt") ,   optional:true, emit: duplicates
```

File: modules/nf-core/genrich/main.nf [hunk 6]
```diff
@@ -8,54 +8,43 @@ process GENRICH {
...
-    tuple val(meta), path("*pileup.bedGraph") , optional:true, emit: bedgraph_pileup
-    tuple val(meta), path("*intervals.bed")   , optional:true, emit: bed_intervals
-    tuple val(meta), path("*duplicates.txt")  , optional:true, emit: duplicates
-    path "versions.yml"                                      , emit: versions
+    tuple val(meta), path("*.narrowPeak")                     , emit: peak
+    path "versions.yml"                                       , emit: versions
+
+    tuple val(meta), path("*.pvalues.bedGraph"), optional:true, emit: bedgraph_pvalues
+    tuple val(meta), path("*.pileup.bedGraph") ,  optional:true, emit: bedgraph_pileup
+    tuple val(meta), path("*.intervals.bed") ,    optional:true, emit: bed_intervals
```
Comment (SPPearce) [inline modules/nf-core/genrich/main.nf:20]:
```suggestion
    tuple val(meta), path("*.intervals.bed")   ,    optional:true, emit: bed_intervals
```

**Review by SPPearce — APPROVED** [review]

**Review by JoseEspinosa — APPROVED** [review]
Awesome! 🚀  
You just need to remove this [line](https://github.com/nf-core/modules/blob/384f4f97f628c0cfd5e1e3d87a9845b1fde6fa3a/tests/modules/nf-core/genrich/test.yml#L38) and lint will pass and we should be ready to merge once all tests pass

**Review by FriederikeHanssen — APPROVED** [review]

File: tests/modules/nf-core/genrich/test.yml [hunk 7]
```diff
@@ -35,11 +35,11 @@
     - path: output/genrich/test.intervals.bed
       md5sum: 4bea65caa3f4043d703af4b57161112e
     - path: output/genrich/test.narrowPeak
-      md5sum: d41d8cd98f00b204e9800998ecf8427e
+      md5sum: 82bc06c2e44e4d91152a6ac6557a2c6e
```
Comment (JoseEspinosa) [inline tests/modules/nf-core/genrich/test.yml:38]: Even better 😍

**Pull request merged**